_posts/2018-12-10-snowplow-javascript-tracker-2.10.0-released/index.md

```diff
@@ -0,0 +1,162 @@
+4. [Updates and bug fixes](#updates)
+5. [Upgrading](#upgrade)
+6. [Documentation and help](#doc)
+
+<!--more-->
+
+<h2 id="global-contexts">1. Global contexts</h2>
+
+This release introduces a set of powerful tracker features for controlling when contexts are added to events. Traditionally, trackers have offered predefined contexts that add contexts to every event when enabled. We refer to these new contexts with expanded capabilities as global contexts because each global context is evaluated when an event is sent.
+
+Global context capabilities:
+
+- adding contexts to every event
+- evaluating contexts dynamically, we allow user-supplied callbacks that optionally accept an argument containing important information (schema URI, event payload, event type)
+- adding contexts based on an event's schema URI
+- adding contexts based on the result of a user-supplied callback
+
+The following methods are added to the tracker:
+1. [`addGlobalContexts`][add-global]
+2. [`removeGlobalContexts`][remove-global]
+3. [`clearGlobalContexts`][clear-global]
+
+For a taste of the functionality, here we can define that a context is only sent for events with the schema vendor `com.acme.marketing`. Notice that `addGlobalContexts` must be supplied with an array of global contexts.
+
+{% highlight javascript %}
+let user_context = {
+  schema: 'iglu:com.acme.contexts/user_context/jsonschema/1-0-0'
+  data: {
+    userid: 1234,
+    name: 'john doe'
+  }
+};
+
+// if a global context has a conditional component (in this case, a schema rule set)
+// our global context must be supplied as an array, where the first element is
+// the conditional part and the context is the second element
+let global_context = [
+  {accept: 'iglu:com.acme.marketing/\*/jsonschema/1-0-0'},
+  user_context
+]
+
+window.snowplow('addGlobalContexts', [global_context]);
+{% endhighlight %}
+
+Full documentation can be found [here][global-contexts-docs].
+
+<h2 id="error-supression">2. Error-handling</h2>
+
+With this version, by default all exceptions generated by the tracker will not surface.
+
+You can disable this by setting the following value:
+
+{% highlight javascript %}
+window.snowplow('addGlobalContexts', [global_context]);
+{% endhighlight %}
+
+<h2 id="link-form-tracking">3. Dynamic contexts for link and form tracking</h2>
+
+
+The feature allows dynamic contexts to be passed in the `context` argument seen in the examples methods below for enabling link and form tracking. For form change events, context generators are passed `(elt, type, value)`, and form submission events are passed `(elt, innerElements)`. Link tracking events pass the source element to the context generator.
+
+{% highlight javascript %}
+window.snowplow('enableLinkClickTracking', criterion, pseudoClicks, trackContent, context);
+window.snowplow('enableFormTracking', config, context);
+{% endhighlight %}
+
+<h2 id="beacon-api">4. Beacon API for event sending</h2>
```

Comment:
For me this would need another line to explain the _what_ and _why_ of the Beacon API. Nothing too deep.